pagopa/io-app
# [IOBP-54] addition of discount-type transactions

## Short description
added timeline transactions and details components for `discount` type initiatives.
| list | pending | canceled |
|--------|--------|--------|
| <img width="200" alt="image" src="https://github.com/pagopa/io-app/assets/60693085/ec33a0c7-a08b-4378-819a-9e627628be5f">| <img width="200" alt="image" src="https://github.com/pagopa/io-app/assets/60693085/b17770f7-941c-467e-a0f8-a22041c6528a"/> | <img width="200" src="https://github.com/pagopa/io-app/assets/60693085/af39d9e5-d37c-41ab-b398-f9a901c07be4"/>| 
## List of changes proposed in this pull request
- addition of required i18n keys
- update of backend idpay specs 
- update in the rendering logic for timelineDetails' listItem
- addition of new component for the transaction's details' bottomSheet
- addition of new required icon for online payments
- updated tests where needed

## How to test
with `io-dev-server` (currently using [this](https://github.com/pagopa/io-dev-api-server/tree/IOBP-54-idpay-discount-initiatives-timeline) branch), and a config like 
```json
"wallet":{
    "idPay":{
        "discountCount":5,
        "refundCount":5,
        "refundNotConfiguredCount":5
    }
}
```
navigate to the wallet screen, then to an initiative labeled with **[D]** (as in "Discount"), and check that all components render correctly.
